.github/workflows/run-tests.yml

```diff
@@ -43,11 +43,11 @@ jobs:
         sudo apt-get update
         sudo apt-get install -y gdb-multiarch python3-dev python3-pip python3-wheel python3-setuptools git cmake gcc g++ pkg-config libglib2.0-dev
 
-    - name: Compile Capstone/Keystone/Unicorn and install Python modules
+    - name: Install Python modules (Capstone, Keystone, Unicorn, Ropper)
       if: steps.cache-libs.outputs.cache-hit != 'true'
       run: |
         mkdir -p ~/.cache/pip || echo ok
-        python3 -c 'import capstone, keystone, unicorn' || python3 -m pip install --user --upgrade -r ./requirements.txt
+        python3 -m pip install --user --upgrade -r ./requirements.txt
```

Comment:
if we pin specific versions in requirements.txt we can get rid of `--upgrade`

Reply:
AFAIK those 3 libs have very stable (and unified) API. I don't see what benefit would come to pinning.

Reply:
We need to check that the lib bindings exist (`python3 -c 'import capstone, keystone, unicorn'`) for the cache to be actionable or not (hence the `if: steps.cache-libs.outputs.cache-hit != 'true'`). See https://github.com/actions/cache/blob/main/examples.md#python---pip

Reply:
I don't think you have that right. The whole `run` key will be skipped if the cache hits. The import is just a waste of time. None of the examples from your link do this import thing. In fact, I think you have the key wrong: The hashfiles should includes _requirements.txt_. What populates **/lockfiles?

Reply:
You were using an old version where lockfiles were used because the cache was created from the compiled version. Now it's correct (https://github.com/hugsy/gef/blob/dev/.github/workflows/run-tests.yml#L44)
Also ok I sent you the link from a wrong tab, see https://github.com/actions/cache#skipping-steps-based-on-cache-hit

Reply:
Anyway yeah I see your point about the `import`, it is useless